### PR TITLE
tests:base:ethernets: Improve stability of autopkgtests

### PR DESCRIPTION
**Pull request by slyon**
Make sure NetworkManager's `unmanaged-devices` are kept in check so that NM does not take control over some network interfaces that are controlled by the test system, such as the routers veth42, veth42, fake wifi AP, ...

Replaces the "autopkgtest-fixes.patch" Distro patch.

Changing the location of the denylist from /run to /etc because the changes in /run would be overwritten by our test system's /etc/NetworkManager/conf.d/90-test-ignore.conf, because /etc/ drop-in configs are read after /run/ drop-ins and the /etc/ drop-in uses a hard overwrite of `unmanaged-devices=...` in order to reset the system's default in /usr/lib/NetworkManager/conf.d/10-globally-managed-devices.conf (guess we could also downgrade the test systems static /etc/ drop-in to /run/ which should have the same effect).

So with this change we would have the following order:
1) base system: (system config, to be overwritten)
/usr/lib/NetworkManager/conf.d/10-globally-managed-devices.conf
2) setUpClass: (static test system config, i.e. management interface eth0, en*)
/etc/NetworkManager/conf.d/90-test-ignore.conf
3) create_devices: (dynamic test interfaces, like eth42, eth43, wlan1, ...)
/etc/NetworkManager/conf.d/99-test-denylist.conf

> [...] you can add additional .conf files to the /etc/NetworkManager/conf.d directory. These will be read in order, with later files overriding earlier ones. Packages might install further configuration snippets to /usr/lib/NetworkManager/conf.d. This directory is parsed first, even before NetworkManager.conf. Scripts can also put per-boot configuration into /run/NetworkManager/conf.d. This directory is parsed second, also before NetworkManager.conf. The loading of a file /run/NetworkManager/conf.d/name.conf can be prevented by adding a file /etc/NetworkManager/conf.d/name.conf. Likewise, a file /usr/lib/NetworkManager/conf.d/name.conf can be shadowed by putting a file of the same name to either /etc/NetworkManager/conf.d or /run/NetworkManager/conf.d.

https://developer.gnome.org/NetworkManager/stable/NetworkManager.conf.html

## Checklist

- [x] Runs `make check` successfully.
- [x] Retains 100% code coverage (`make check-coverage`).
- [ ] New/changed keys in YAML format are documented.
- [ ] \(Optional\) Adds example YAML for new feature.
- [ ] \(Optional\) Closes an open bug in Launchpad.

### CODE REVIEW

File: tests/integration/base.py [hunk 1]
```diff
@@ -444,8 +445,8 @@ def create_devices(klass):
         klass.dev_w_client = devs[1]
 
         # don't let NM trample over our fake AP
-        with open('/run/NetworkManager/conf.d/test-blacklist.conf', 'w') as f:
-            f.write('[main]\nplugins=keyfile\n[keyfile]\nunmanaged-devices+=nptestsrv,%s\n' % klass.dev_w_ap)
+        with open('/etc/NetworkManager/conf.d/99-test-denylist.conf', 'w') as f:
```
Comment (schopin-pro) [inline tests/integration/base.py:448]: Out of curiosity, why the change from /run to /etc ?

**Comment by slyon** [comment]
This PR should be integrated with https://git.launchpad.net/~ubuntu-core-dev/netplan/+git/ubuntu/commit/?id=abdfe5deb62ed312dec56a67ae4dae42a348fbf1

File: tests/integration/base.py (hunk 1 above)
Comment (slyon) [reply tests/integration/base.py:448]: This is because the changes in /run would be overwritten by our test system's /etc/NetworkManager/conf.d/90-test-ignore.conf, because /etc/ drop-in configs are read after /run/ drop-ins and the /etc/ drop-in uses a hard overwrite of `unmanaged-devices=...` in order to reset the system's default in /usr/lib/NetworkManager/conf.d/10-globally-managed-devices.conf (guess we could also downgrade the test systems static /etc/ drop-in to /run/ which should have the same effect).

So with this change we would have the following order:
1) base system: (system config, to be overwritten)
/usr/lib/NetworkManager/conf.d/10-globally-managed-devices.conf
2) setUpClass: (static test system config, i.e. management interface eth0, en*)
/etc/NetworkManager/conf.d/90-test-ignore.conf
3) create_devices: (dynamic test interfaces, like eth42, eth43, wlan1, ...)
/etc/NetworkManager/conf.d/99-test-denylist.conf

> [...] you can add additional .conf files to the /etc/NetworkManager/conf.d directory. These will be read in order, with later files overriding earlier ones. Packages might install further configuration snippets to /usr/lib/NetworkManager/conf.d. This directory is parsed first, even before NetworkManager.conf. Scripts can also put per-boot configuration into /run/NetworkManager/conf.d. This directory is parsed second, also before NetworkManager.conf. The loading of a file /run/NetworkManager/conf.d/name.conf can be prevented by adding a file /etc/NetworkManager/conf.d/name.conf. Likewise, a file /usr/lib/NetworkManager/conf.d/name.conf can be shadowed by putting a file of the same name to either /etc/NetworkManager/conf.d or /run/NetworkManager/conf.d.

https://developer.gnome.org/NetworkManager/stable/NetworkManager.conf.html

**Comment by slyon** [comment]
I conducted some PPA autopkgtest testing to confirm those fixes. It's looking good and I'll merge it.

* https://autopkgtest.ubuntu.com/results/autopkgtest-lunar-slyon-testing/lunar/amd64/n/netplan.io/20230208_132926_35ddf@/log.gz
* https://autopkgtest.ubuntu.com/results/autopkgtest-lunar-slyon-testing/lunar/armhf/n/netplan.io/20230208_115551_087de@/log.gz
* https://autopkgtest.ubuntu.com/results/autopkgtest-lunar-slyon-testing/lunar/ppc64el/n/netplan.io/20230208_115910_124b3@/log.gz
* https://autopkgtest.ubuntu.com/results/autopkgtest-lunar-slyon-testing/lunar/s390x/n/netplan.io/20230208_130047_f7986@/log.gz
